File: src/main/java/com/netcracker/models/ApartmentSubBill.java
package com.netcracker.models;

import lombok.Data;

import java.math.BigInteger;
import java.util.List;

@Data
public class ApartmentSubBill extends SubBill {
    private Double debt;
    private Apartment apartment;
    private List<ApartmentOperation> apartmentOperations;
    private List<DebtPaymentOperation> debtPaymentOperations;

    public ApartmentSubBill(BigInteger subBill, Double balance, Double debt, List<ApartmentOperation> apartmentOperation,
                            List<DebtPaymentOperation> debtPaymentOperation, CommunalUtility communalUtility,Apartment apartment) {
        super(subBill, balance,communalUtility);
        this.debt = debt;
        this.apartmentOperations = apartmentOperation;
        this.debtPaymentOperations = debtPaymentOperation;
        this.apartment=apartment;
    }

    public ApartmentSubBill() {
    }
}

File: src/main/angular/app/models/apartment-sub-bill.ts
import {CommunalUtility} from "./communal-utility";
import {Apartment} from "./apartment";
import {ApartmentOperation} from "./apartment-operation";
import {DebtPaymentOperation} from "./debt-payment-operation";

export class ApartmentSubBill{
    subBillId?: number;
    balance?: number;
    communalUtility?: CommunalUtility;
    debt?: number;
    apartment: Apartment;
    apartmentOperations?: ApartmentOperation[];
    debtPaymentOperations?: DebtPaymentOperation[];
}